spacy/cli/apply.py

```diff
@@ -0,0 +1,123 @@
+import tqdm
+import srsly
+
+from itertools import chain
+from pathlib import Path
+from typing import Optional, List, Iterable, cast, Union
+
+from wasabi import msg
+
+from ._util import app, Arg, Opt, setup_gpu, import_code, walk_directory
+
+from ..tokens import Doc, DocBin
+from ..vocab import Vocab
+from ..util import ensure_path, load_model
+
+
+path_help = ("Location of the documents to predict on."
+             "Can be a single file in .spacy format or "
+             "a text file with one document per line."
+             "If a directory is provided each "
+             "text file in the directory will be treated "
+             "as a single document.")
+out_help = "Path where to save the result .spacy file"
+code_help = ("Path to Python file with additional "
+             "code (registered functions) to be imported")
+gold_help = "Use gold preprocessing provided in the .spacy files"
+
+DocOrStrStream = Union[Iterable[str], Iterable[Doc]]
+
+
+def _stream_docbin(path: Path, vocab: Vocab) -> Iterable[Doc]:
+    """
+    Stream Doc objects from DocBin.
+    """
+    docbin = DocBin().from_disk(path)
+    for doc in docbin.get_docs(vocab):
+        yield doc
+
+
+def _stream_jsonl(path: Path) -> Iterable[str]:
+    """
+    Stream "text" field from JSONL. If the field "text" is
+    not found it raises error.
+    """
+    for entry in srsly.read_jsonl(path):
+        if "text" not in entry:
+            raise ValueError(
+                f"{path} does not contain the required 'text' field."
+            )
+        else:
+            yield entry["text"]
+
+
+def _stream_texts(paths: Iterable[Path]) -> Iterable[str]:
+    """
+    Yields strings from text files in paths.
+    """
+    for path in paths:
+        with open(path, 'r') as fin:
+            text = fin.read()
+            yield text
+
+
+@app.command("apply")
+def apply_cli(
+    # fmt: off
+    model: str = Arg(..., help="Model name or path"),
+    data_path: Path = Arg(..., help=path_help, exists=True),
+    output: Path = Arg(..., help=out_help, dir_okay=False),
```

Comment:
I would rename this `output_file` to make it clear it's always a file and never a directory